### PR TITLE
#226 fix for unit testing

### PR DESCRIPTION
**Pull request by mmilitaru**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [ ] I have provided test coverage for my change (where applicable)

### Description
#226 Unit test fixes upon context menu modification

### CODE REVIEW

File: EngineeringModel.Tests/ViewModels/FiniteStateBrowser/FiniteStateBrowserViewModelTestFixture.cs [hunk 1]
```diff
@@ -192,8 +192,17 @@ public void VerifyThatContextMenuPopulated()
             this.iteration.ActualFiniteStateList.Add(actualList);
 
             var viewmodel = new FiniteStateBrowserViewModel(this.iteration, this.session.Object, this.thingDialogNavigationService.Object, this.panelNavigationService.Object, null, null);
-            
-            // no row selected
```
Comment (lxatrhea) [inline EngineeringModel.Tests/ViewModels/FiniteStateBrowser/FiniteStateBrowserViewModelTestFixture.cs]: "no row selected" scenario is not in the test anymore

File: EngineeringModel.Tests/ViewModels/FiniteStateBrowser/FiniteStateBrowserViewModelTestFixture.cs [hunk 2]
```diff
@@ -192,8 +192,17 @@ public void VerifyThatContextMenuPopulated()
...
+            viewmodel.FiniteStateList.Add(new CDP4Composition.FolderRowViewModel("Actual List", "Possible Finite State List", viewmodel.Session, viewmodel));
```
Comment (lxatrhea) [inline EngineeringModel.Tests/ViewModels/FiniteStateBrowser/FiniteStateBrowserViewModelTestFixture.cs]: Adding viewmodels is not necessary. It is done in the Initialize method.

File: EngineeringModel.Tests/ViewModels/FiniteStateBrowser/FiniteStateBrowserViewModelTestFixture.cs (hunk 1 above)
Comment (mmilitaru) [reply EngineeringModel.Tests/ViewModels/FiniteStateBrowser/FiniteStateBrowserViewModelTestFixture.cs]: No row selected was actually selected row of one of the two options, which are being covered by the cases. It refereed to rows that are not children.

Comment (alexatrhea) [reply EngineeringModel.Tests/ViewModels/FiniteStateBrowser/FiniteStateBrowserViewModelTestFixture.cs]: You can still add an assertion when `SelectedThing = null`.

Comment (lxatrhea) [reply EngineeringModel.Tests/ViewModels/FiniteStateBrowser/FiniteStateBrowserViewModelTestFixture.cs]: agreed

File: EngineeringModel.Tests/ViewModels/FiniteStateBrowser/FiniteStateBrowserViewModelTestFixture.cs (hunk 2 above)
Comment (mmilitaru) [reply EngineeringModel.Tests/ViewModels/FiniteStateBrowser/FiniteStateBrowserViewModelTestFixture.cs]: That is correct. Code removed.

**Comment by mmilitaru** [comment]
Additional testing was added for correct  menu verification.

File: EngineeringModel.Tests/ViewModels/FiniteStateBrowser/FiniteStateBrowserViewModelTestFixture.cs (hunk 1 above)
Comment (mmilitaru) [reply EngineeringModel.Tests/ViewModels/FiniteStateBrowser/FiniteStateBrowserViewModelTestFixture.cs]: I do not think that is necessary. SelectedThing cannot be null. There are always 2 rows created in initialization. In my opinion that check is redundant.

Comment (mmilitaru) [reply EngineeringModel.Tests/ViewModels/FiniteStateBrowser/FiniteStateBrowserViewModelTestFixture.cs]: The check for null selection has been added.

**Pull request merged**